src/Data/Yaml/Internal.hs

```diff
@@ -125,15 +130,21 @@ instance Monad m => Monad (PErrorT m) where
 instance MonadTrans PErrorT where
     lift = PErrorT . liftM Right
 instance MonadIO m => MonadIO (PErrorT m) where
-    liftIO = lift . liftIO
+    liftIO = liftIO
```

Comment:
This looks like an infinite loop.

Reply:
Nice catch!

Apparently this is dead code.  I removed it in https://github.com/snoyberg/yaml/pull/146/commits/8d491e4772f6678208b5fd0ee3e8180788a7a751.